packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,630 @@
+import {
+  Diagnostic,
+  DiagnosticCollector,
+  NoTarget,
+  SourceFile,
+  createDiagnosticCollector,
+  resolvePath,
+} from "@typespec/compiler";
+import { HttpStatusCodeRange } from "@typespec/http";
+import { resolveOperationId } from "@typespec/openapi";
+import {
+  SdkAnyExample,
+  SdkArrayExample,
+  SdkArrayType,
+  SdkBodyModelPropertyType,
+  SdkClientType,
+  SdkDictionaryExample,
+  SdkDictionaryType,
+  SdkHttpOperation,
+  SdkHttpOperationExample,
+  SdkHttpParameter,
+  SdkHttpParameterExample,
+  SdkHttpResponse,
+  SdkHttpResponseExample,
+  SdkModelExample,
+  SdkModelPropertyType,
+  SdkModelType,
+  SdkNullExample,
+  SdkServiceMethod,
+  SdkServiceOperation,
+  SdkType,
+  SdkTypeExample,
+  SdkUnionExample,
+  isSdkFloatKind,
+  isSdkIntKind,
+} from "./interfaces.js";
+import { TCGCContext } from "./internal-utils.js";
+import { createDiagnostic } from "./lib.js";
+
+interface LoadedExample {
+  readonly relativePath: string;
+  readonly file: SourceFile;
+  readonly data: any;
+}
+
+async function loadExamples<TServiceOperation extends SdkServiceOperation>(
+  context: TCGCContext,
+  client: SdkClientType<TServiceOperation>
+): Promise<[Map<string, Record<string, LoadedExample>>, readonly Diagnostic[]]> {
+  const diagnostics = createDiagnosticCollector();
+  if (!context.examplesDirectory) {
+    return diagnostics.wrap(new Map());
+  }
+
+  const allApiVersions = client.apiVersions;
+  let apiVersion = context.apiVersion;
+  if (apiVersion === "all") {
+    return diagnostics.wrap(new Map());
+  }
+  if (apiVersion === "latest" || apiVersion === undefined || !allApiVersions.includes(apiVersion)) {
```

Comment:
can we make this a helper function, since I think we're doing similar apiVersion checks throughout the code